File: javascript/168.excel-sheet-column-title.js
/*
 * @lc app=leetcode id=168 lang=javascript
 *
 * [168] Excel Sheet Column Title
 */

// @lc code=start
/**
 * @param {number} columnNumber
 * @return {string}
 */
var convertToTitle = function (columnNumber) {
  let columnTitle = "";
  while (columnNumber > 0) {
    columnNumber -= 1;
    columnTitle = String.fromCharCode(65 + (columnNumber % 26)) + columnTitle;
    columnNumber = ~~(columnNumber / 26);
  }
  return columnTitle;
};
// @lc code=end

File: rust/168.excel-sheet-column-title.rs
/*
 * @lc app=leetcode id=168 lang=rust
 *
 * [168] Excel Sheet Column Title
 */

// @lc code=start
impl Solution {
    pub fn convert_to_title(column_number: i32) -> String {
        let mut result = String::new();
        let mut i = column_number;
        while i > 0 {
            i -= 1;
            result.insert(0, (i % 26 + 65) as u8 as char);
            i /= 26;
        }
        return result;
    }
}
// @lc code=end
